August Heartbeat Blog post (#3876)
iterative/dvc.org: content/blog/2022-08-16-august-22-heartbeat.md

## Vanishing Gradients Podcast

![Vanishing Gradients](/uploads/images/2022-08-16/vanishing-gradients.png 'Vanishing Gradients :wrap-left =300')
If you are not familiar with
[**Hugo Bowne-Anderson**](https://twitter.com/hugobowne), you should be. He was
the host of my all-time favorite Data Science podcast
[DataFramed](https://www.datacamp.com/podcast) while he was at
[DataCamp](https://www.datacamp.com/). DataFramed helped me immeasurably when I
started my data science journey. It provided great not only great teachings on
many data science concepts, but even more importantly, the ability to gain
perspectives from different people across all parts of the data space, talking
about challenges, danger zones, and issues that we all need to be aware of in
the field. Recently Hugo started a new podcast,
[Vanishing Gradients](https://vanishinggradients.fireside.fm/). This newer
endeavor is in a somewhat different format than DataFramed, but still with
Hugo's characteristic deep dive into all the challenges that come up when
working with data. Hugo uses a long-format conversation approach with many
leaders and great thinkers in the data science/machine learning/AI space. In

> episodes, [seven](https://vanishinggradients.fireside.fm/7) and
> [eight](https://vanishinggradients.fireside.fm/8) Hugo has a fascinating chat

Reviewer: ```suggestion
episodes [seven](https://vanishinggradients.fireside.fm/7) and
[eight](https://vanishinggradients.fireside.fm/8), Hugo has a fascinating chat
```

You can find more about his workflow in
[his repo for the project.](https://github.com/mikolajkania/kaggle-03-house-prices)

![Using DVC for Kaggle Competition](/uploads/images/2022-08-16/kaggle-dvc.png '=800')
_DVC with Kaggle
([Source link](https://mikolajkania.com/2022/08/07/dvc-kaggle-mlops/))_

Mikołaj explains how DVC's project structure ensures reproducible results and
develops habits on best practices. One drawback he noted was the lack of an
experimentation UI, but we just introduced the DVC extension for VS Code to help
with that, and there’s always Iterative Studio. Look out for improvement to the
experiment features in both tools in the coming months! Also, experimenting with
DVC in Kaggle may give you some good practice for things we are cooking up
internally! 😉🤫

## **Shambhavi Mishra** - Searching for Semantic Similarity

[**Shambhavi Mishra**](https://twitter.com/ShambhaviCodes) in her post
[Searching for Semantic Similarity](https://medium.com/towards-artificial-intelligence/searching-for-semantic-similarity-cfbff2388d04)
details the steps of her NLP project on similarity algorithms. She mainly

> focuses on cosine similarity using a Stackoverflow questions dataset. The

Reviewer: ```suggestion
focuses on cosine similarity using a Stack Overflow questions dataset. The
```

([Source link](https://www.oreilly.com/library/view/mastering-machine-learning/9781785283451/ba8bef27-953e-42a4-8180-cea152af8118.xhtml))_

## **Evgenii Munin** - Run S3 Locally With MinIO for the DVC Machine Learning Pipeline

If you are in need of object storage to work with data through an API, but need
to do so in a private network,
[**Evgenii Munin**](https://www.linkedin.com/in/evgenii-munin-01932a143/) shows
how to set up MinIO as remote storage with DVC to do just that
[in this piece in Medium](https://betterprogramming.pub/run-s3-locally-with-minio-for-dvc-machine-learning-pipeline-7fa3d240d3ab).
In this cool use case, he starts with installing the MinIO server and builds a
Docker image to run it, sharing a great repo on Kafka-to S3 where MinIO was used
to mock the S3 for the data. Then he shows you how to link the MinIO server as
DVC remote storage.

![Minio Browser](/uploads/images/2022-08-16/minio.png '=800') _Minio Browser
with Data pushed from DVC
([Source link](https://betterprogramming.pub/run-s3-locally-with-minio-for-dvc-machine-learning-pipeline-7fa3d240d3ab))_

## **Caleb Kaiser** - Moving from Data Science to Machine Learning Engineering

> It can sometimes be confusing to determine where data science stops and machine

Reviewer: Could you break this up into a couple of paragraphs? It'll break up the wall of text.

learning engineering starts. [**Caleb Kaiser**](https://twitter.com/KaiserFrose)
helps clarify this
[in this old but good piece](https://www.kdnuggets.com/2020/11/moving-data-science-machine-learning-engineering.html)
in [KD Nuggets](https://www.kdnuggets.com). He provides four examples of real-
world projects and defines what portions of the project are data science and
what are ML engineering. In all what we find is that machine learning
engineering is all the tasks that need to happen to get the model the data
scientists create into production applications. He goes on to dive deeper into
one of the examples and shows the promise in some tools that bridge the gap
between machine learning and software engineering where he highlights DVC and
Huggingface. This is a good piece to read if you are grappling with the
difference!

https://media.giphy.com/media/xUNd9DLukkavmhybAs/giphy.gif

## Just a few other things...

> - Github Goodness alert for

Reviewer: ```suggestion
- GitHub Goodness alert for
```

Double-check the spelling throughout.

- If you are interested in contributing to researchers' learning about machine
  learning experimentation tools, you can take
  [this survey](https://www.freelancer.com.au/projects/machine-learning/Seeking-Qualified-Respondents-for-Online-34294453.html).
  Spread the word!

## Company News

### 🎉 Model Registry released in Iterative Studio

On July 26th we released our new
[model registry in Iterative Studio.](https://iterative.ai/model-registry)  
The great work done by the MLEM team building a git-based model registry is now
incorporated in Studio in a web UI. This release took the work of half the
people in the company and we are proud of the steps we are taking to meet people
where they are and round out your options whether you are comfortable in the
CLI, API, or web UI. Be sure to try it out and give us your feedback. Learn more
[in the blog post](https://dvc.org/blog/iterative-studio-model-registry) and
[in the docs](https://dvc.org/doc/studio/user-guide/model-registry/what-is-a-model-registry).
Look out for a full tutorial coming soon!

https://www.youtube.com/watch?v=DYeVI-QrHGI

### 🧑🏽‍💻 Iterative's First Internal Hackathon

> Last week we had our very first internal Hackathon! The entire company

Reviewer: Break this up into a couple of paragraphs.
